lib/jsrt/JsrtSourceHolder.cpp

```diff
@@ -33,8 +33,10 @@ namespace Js
             Js::JavascriptError::ThrowOutOfMemoryError(nullptr);
         }
 
-        size_t cbUtf8Buffer = (length + 1) * 3;
-        if (cbUtf8Buffer > UINT_MAX)
+        // `length` should not be bigger than MAXLONG 
+        // UINT_MAX / 3 < MAXLONG
+        size_t cbUtf8Buffer = (UINT_MAX / 3 > length + 1) ? (length + 1) * 3 : UINT_MAX;
```

Comment:
let's make it more explicit? ((UINT_MAX / 3 - 1) > length) ? (length + 1) * 3 : UINT_MAX
good catch